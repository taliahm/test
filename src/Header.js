import React, { useEffect, useState } from "react";

function Header() {
  const [loggedIn, setLoggedIn] = useState(true);
  const [name, setName] = useState('');
  useEffect(() => {
    async function fetchPoke() {
      const response = await fetch('https://pokeapi.co/api/v2/pokemon/ditto');
      const json = await response.json();
      setName(json.name);
    }
    fetchPoke();
  }, [])
  return (
    <header>
      <h1>Pizza Party Plaza</h1>
      <button onClick={() => {setLoggedIn(!loggedIn)}}>
        { loggedIn ? 'Logout' : 'Login' }
      </button>
      <p data-testid="name">welcome {name}</p>
    </header>
  );
}

export default Header;