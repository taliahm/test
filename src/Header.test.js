import { render, fireEvent, screen, waitFor } from "@testing-library/react";
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom/extend-expect'
import Header from "./Header";

describe('Header component', () => {
    afterEach(() => {    
      jest.clearAllMocks();
    });
    it('rendering ability', () => {
        // Arrange
        const { getByText } = render(<Header />);
        getByText('Pizza Party Plaza');
        const button = getByText("Logout");

        // Act
        // fireEvent.click(button);
        userEvent.click(button);
        screen.logTestingPlaygroundURL()
        // Assert
        // expect(button.textContent).toMatch("Login");
        expect(button).toHaveTextContent("Login");
    })
    it ('the ability to fetch a name', async () => {
        const spy = jest.spyOn(global, "fetch").mockImplementation(() =>
        Promise.resolve({
          json: () => Promise.resolve({name: "steve"})
        })
      );

        const { getByTestId } = render(<Header />);


        const p = getByTestId("name");
        await waitFor(() => expect(p).toHaveTextContent("welcome steve"))
        expect(spy).toHaveBeenCalled();
        spy.mockRestore()
        // expect(p).toHaveTextContent("welcome steve")
    })
    test("header renders properly", () => {
        const {asFragment} = render(<Header special="Turnip Pizza" />);
        
        const renderedHtml = asFragment();
      
        expect(renderedHtml).toMatchSnapshot();
      });
})