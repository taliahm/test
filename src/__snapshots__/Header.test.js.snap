// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component header renders properly 1`] = `
<DocumentFragment>
  <header>
    <h1>
      Pizza Party Plaza
    </h1>
    <button>
      Logout
    </button>
    <p
      data-testid="name"
    >
      welcome 
    </p>
  </header>
</DocumentFragment>
`;
